### ISSUE TITLE
Fix netbox database

### ISSUE THREAD

**Issue by naved001** [CCI-MOC/ops-issues#632]
> [[FIRING:1] netbox (PersistentVolumeUsageCritical https-metrics 128.52.62.215:10250 kubelet /metrics oct5-12-compute.ocp-prod.massopen.cloud netbox-pgha1-jvcn-pgdata openshift-monitoring/k8s openshift-storage.rbd.csi.ceph.com kubelet critical ocs-external-storagecluster-ceph-rbd)](https://alertmanager-main-openshift-monitoring.apps.ocp-prod.massopen.cloud/#/alerts?receiver=Critical)

Because it is getting full. But before that I need to test if I can expand volumes without having to modify the storage class.

https://docs.openshift.com/container-platform/4.10/storage/expanding-persistent-volumes.html

**Comment by naved001** [comment]
looks like I can expand a volume without modifying the storage class.

**Comment by naved001** [comment]
> looks like I can expand a volume without modifying the storage class.

the parameter is actually set, I just missed it.

```
naved@Naveds-MacBook-Pro office-stuff % oc get sc ocs-external-storagecluster-ceph-rbd -o json | jq .allowVolumeExpansion
true
naved@Naveds-MacBook-Pro office-stuff % oc get sc ocs-external-storagecluster-cephfs -o json | jq .allowVolumeExpansion
true
```

**Comment by naved001** [comment]
@hakasapl netbox is down now :(

**Comment by naved001** [comment]
both databases are stuck in starting state

```
bash-4.4$ patronictl list
+---------------------+---------------------------------+---------+----------+----+-----------+
| Member              | Host                            | Role    | State    | TL | Lag in MB |
+ Cluster: netbox-ha (7073218847275253843) -------------+---------+----------+----+-----------+
| netbox-pgha1-bc2z-0 | netbox-pgha1-bc2z-0.netbox-pods | Replica | starting |    |   unknown |
| netbox-pgha1-jvcn-0 | netbox-pgha1-jvcn-0.netbox-pods | Replica | starting |    |   unknown |
+---------------------+---------------------------------+---------+----------+----+-----------+
```

**Comment by knikolla** [comment]
We did some debugging during a Zoom call (with Naved and Rob). My knowledge of HA Postgres on Kubernetes is very limited, and typing commands randomly didn't seem to magically fix things. Attempted to reduce the number of replicas to 1, then do a failover to the only remaining replica using `patronictl failover` but that didn't seem to work, still complaining about no available good candidates. 

Last good backup is from July 5.

**Comment by naved001** [comment]
https://github.com/CCI-MOC/ops-issues/issues/643

Quoting @larsks here who ended up fixing this.

> I think the primary problem was the expired certificates on the rgw servers. Need to revisit that process -- certbot is working correctly, but haproxy wasn't restarted last time the certificate was re-provisioned.

> That's why backups stopped working, and backups not working is probably why the wal directories filled up.

> With the certificates fixed, I was able to follow the instructions in https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/disaster-recovery/ to perform a recovery from backups.

**Comment by naved001** [comment]
@larsks This issue has cropped up again, the netbox PV is running out of space, reated to the backups not working since there are a bunch of failed backups.

**Comment by naved001** [comment]
the certificate that HAproxy was presenting expired on Sep 4, so I restarted HAproxy which then picked up the renewed certificate that was available.

**Comment by naved001** [comment]
a backup was successfully completed now, so will the wal files gradually clear up then?
```
netbox-pgbackrest-repo1-incr-27716657--1-xg47n   0/1     Completed   0          12m
```

**Comment by naved001** [comment]
the backups are being completed successfully, but not sure what would it take for the wal files to clear up. 

```
+---------------------+---------------------------------+---------+---------+----+-----------+
bash-4.4$ patronictl  list
+---------------------+---------------------------------+---------+---------+----+-----------+
| Member              | Host                            | Role    | State   | TL | Lag in MB |
+ Cluster: netbox-ha (7073218847275253843) -------------+---------+---------+----+-----------+
| netbox-pgha1-bc2z-0 | netbox-pgha1-bc2z-0.netbox-pods | Leader  | running | 21 |           |
| netbox-pgha1-qwsh-0 | netbox-pgha1-qwsh-0.netbox-pods | Replica | running | 20 |       624 |
+---------------------+---------------------------------+---------+---------+----+-----------+
```

The Leader is using only 3% percent disk space, while the replica is using 100%.  And looks like the lag is increasing. @larsks any advice on what to do next?

**Comment by naved001** [comment]
tl;dr things look good now. I ran `patronictl reinint netbox-ha netbox-pgha1-qwsh-0` and that reinitialized the problem member (replica).

On the pod `netbox-pgha1-qwsh-0` which was starting to fail, I copied the `/pgdata/pg13_wal` directory to `/tmp/something` and then my plan was to delete the data in `/pgdata/pg13_wal` and create a symbolic link from `/tmp/something/` to that, but apparently when I removed the files from `/pgadata/pg13_wal` the container exited. 
Since I had copied the `pg13_wal` locally, I started uploading it to the respawned `netbox-pgha1-qwsh-0` but the container exited again unexpectedly. Anyway this trick didn't work so I ended up re-initializing the cluster member using patronictl and now it reports that the leader and replica are both running and have plenty of space.

```
bash-4.4$ patronictl list
+---------------------+---------------------------------+---------+---------+----+-----------+
| Member              | Host                            | Role    | State   | TL | Lag in MB |
+ Cluster: netbox-ha (7073218847275253843) -------------+---------+---------+----+-----------+
| netbox-pgha1-bc2z-0 | netbox-pgha1-bc2z-0.netbox-pods | Replica | running | 22 |         0 |
| netbox-pgha1-qwsh-0 | netbox-pgha1-qwsh-0.netbox-pods | Leader  | running | 22 |           |
+---------------------+---------------------------------+---------+---------+----+-----------+
```

I have also created a 1 Gig dummy file `/pgdata/DELETE_IF_POSTGRES_EMERGENCY` on both PVs in case we run into some other out-of-disk-space issues in the future haha.

**Comment by naved001** [comment]
@larsks here we go again...

```
naved@computer ~ % oc logs netbox-pgbackrest-repo2-incr-27800963--1-ktlkz
time="2022-11-10T05:29:59Z" level=info msg="crunchy-pgbackrest starts"
time="2022-11-10T05:29:59Z" level=info msg="debug flag set to false"
time="2022-11-10T05:29:59Z" level=info msg="backrest backup command requested"
time="2022-11-10T05:29:59Z" level=info msg="command to execute is [pgbackrest backup --stanza=db --repo=2 --type=incr]"
time="2022-11-10T05:31:59Z" level=info msg="output=[]"
time="2022-11-10T05:31:59Z" level=info msg="stderr=[ERROR: [095]: unable to load info file '/pgbackrest/repo2/backup/db/backup.info' or '/pgbackrest/repo2/backup/db/backup.info.copy':\n       CryptoError: unable to verify certificate presented by 'rgw-j-vip.int.massopen.cloud:443': [10] certificate has expired\n       HINT: is or was the repo encrypted?\n       CryptoError: unable to verify certificate presented by 'rgw-j-vip.int.massopen.cloud:443': [10] certificate has expired\n       HINT: is or was the repo encrypted?\n       HINT: backup.info cannot be opened and is required to perform a backup.\n       HINT: has a stanza-create been performed?\n]"
time="2022-11-10T05:31:59Z" level=fatal msg="command terminated with exit code 95"
```

**Comment by naved001** [comment]
okay, it failed again because haproxy did not restart. Now I need to figure out why it didn't do that. I remember taking care of it in https://github.com/CCI-MOC/ops-issues/issues/643
